### PR TITLE
[LUPEYALPHA-1075] Early Years Practitioner mobile number

### PR DESCRIPTION
**Pull request by alkesh**
* added mobile number page for EY practitioner journey
* fixed 'resend passcode' functionality for both email and mobile
* fixed bug - if claimant changed answer to "Would you like to provide your mobile number?" to "No" on the check-your-answsers screen, we would still incorrectly save the mobile number.

### CODE REVIEW

File: app/forms/mobile_number_form.rb [hunk 1]
```diff
@@ -15,7 +15,6 @@ class MobileNumberForm < Form
 
   def save
     return false unless valid?
-    return true unless mobile_number_changed?
```
Comment (alkesh) [inline app/forms/mobile_number_form.rb:18]: I'm pretty sure this line meant that the 'resend passcode' functionality never worked.
Also, I can delete this line, and the spec still passes with no changes.

File: app/views/claims/_one_time_password.html.erb [hunk 2]
```diff
@@ -27,9 +27,9 @@
 
       <div class="govuk-!-margin-bottom-6">
         <% if email_or_mobile == "email" %>
-          <%= govuk_link_to "Resend passcode (you will be sent back to the email address page)", claim_path(current_journey_routing_name, "email-address"), no_visited_state: true %>
+          <%= govuk_link_to "Resend passcode (you will be sent back to the email address page)", claim_path(current_journey_routing_name, "email-address", resend: true), no_visited_state: true %>
```
Comment (rjlynch) [inline app/views/claims/_one_time_password.html.erb:30]:
```suggestion
          <%= govuk_link_to "Resend passcode (you will be sent back to the email address page)", claim_path(current_journey_routing_name, "email-address", claim: { resend: true }), no_visited_state: true %>
```

Does setting the resend params like this work and let us avoid having to set the `@resend` instance variable? If so we can lose the callbacks for setting `@resend` and not need to use `@resend` in the view.

**Review by rjlynch — APPROVED** [review]
LGTM but probably worth seeing if we can remove the `@resend` instance variable and lose the two callbacks - not a blocker but I think it should be pretty straightforward

File: app/views/claims/_one_time_password.html.erb (hunk 2 above)
Comment (alkesh) [reply app/views/claims/_one_time_password.html.erb:30]: Using `params[:resent]` in the view avoids the need for the callbacks. 
I couldn't get `claim: { resend: true }` to work here - but I'll try a couple of other things before giving up.

**Pull request merged**